pallets/did/src/migrations.rs

```diff
@@ -0,0 +1,166 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2021 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use codec::{Decode, Encode};
+use kilt_primitives::VersionMigratorTrait;
+use sp_std::marker::PhantomData;
+
+use crate::*;
+
+mod v1;
+
+/// Storage version of the DID pallet.
+#[derive(Copy, Clone, Encode, Eq, Decode, Ord, PartialEq, PartialOrd)]
+pub enum DidStorageVersion {
+	V1,
+	V2,
+}
+
+#[cfg(feature = "try-runtime")]
+impl DidStorageVersion {
+	/// The latest storage version.
+	fn latest() -> Self {
+		Self::V2
+	}
+}
+
+// All nodes will default to this, which is not bad, as in case the "real"
+// version is a later one (i.e. the node has been started with already the
+// latest version), the migration will simply do nothing as there's nothing in
+// the old storage entries to migrate from.
+//
+// It might get updated in the future when we know that no node is running this
+// old version anymore.
+impl Default for DidStorageVersion {
+	fn default() -> Self {
+		Self::V1
+	}
+}
+
+impl<T: Config> VersionMigratorTrait<T> for DidStorageVersion {
```

Comment:
I will try to make this version migration more generic in the next week or future. But since we should wait for https://github.com/paritytech/substrate/pull/9165 to be available in a release, it might not make sense to start with this before.

Reply:
Yes please, I think that would be great as we will eventually need it for other pallets as well. I can also take care of this if you don't manage to do it before you leave.